--- keyhook/vietkey.cpp
+++ keyhook/vietkey.cpp
@@ -642,14 +642,10 @@
 
 	SHORT lower;
-	SHORT shift, capital;
-	shift = GetKeyState(VK_SHIFT);
-	capital = GetKeyState(VK_CAPITAL);
-	if (((shift & 0x8000) && (capital & 0x0001)) ||
-		((shift & 0x8000) == 0 && (capital & 0x00001) == 0))
-		lower = 1;
-	else
-		lower = 0;
+	if (isupper(c) || c == '{' || c == '}') {
+		lower = 0;
+	} else {
+		lower = 1;
+	}
 
 
 	unsigned char newChar;
